### PR TITLE
Add FXIOS-4785 [v106] Adds metrics to track number of history visits

### PR DESCRIPTION
**Pull request by tarikeshaq**
# [FXIOS-4785](https://mozilla-hub.atlassian.net/browse/FXIOS-4785) https://github.com/mozilla-mobile/firefox-ios/issues/11635
Ahead of migrating the history storage to use the application-services places component, we want to track down the number of history visits in a user's database. This adds those metrics.

~~## Keeping as a draft until I get data review~~

I'll probably want a quick glean eyes on this as well to make sure it's OK to add the new metrics.yaml file

### CODE REVIEW

File: Storage/SQL/SQLiteHistory.swift
Comment (tarikeshaq) [inline Storage/SQL/SQLiteHistory.swift:154]: not a fan of running a query purely for the purpose of the counting.. sounds like wasful CPU/IO resources. I'll quickly scan if there are other places I could get the count.

That said, this only happens once, asynchronously on initialization so it should be okay

**Comment by tarikeshaq** [comment]
Data review request in https://github.com/mozilla-mobile/firefox-ios/issues/11635

**Comment by lmarceau** [comment]
Build is green [here](https://app.bitrise.io/app/6c06d3a40422d10f/pipelines/c5a52b5a-7f87-4b5c-9731-4ebb11792bec)
Screen Shot 2022-08-25 at 1 25 33 PM

**Comment by tarikeshaq** [comment]
@nbhasin2 I think this is ready for review, can you think of anything going wrong if we run a query like this? it's asynchronous and runs only once on initialization - I couldn't find another easy spot to collect this telemetry 😕

**Comment by tarikeshaq** [comment]
I also want to cc @travis79 here - No need for you to take time to review the swift changes, but I wanted to make sure that adding an additional `metrics.yaml` like was done here is OK and that telemetry will be gathered properly

**Comment by nbhasin2** [comment]
I am going to close and re-open the issue as BR is being weird for this PR

File: Storage/SQL/SQLiteHistory.swift
Comment (nbhasin2) [inline Storage/SQL/SQLiteHistory.swift:173]: @tarikeshaq is this a method we plan on keeping for long or the idea is to get rid of it once you have the required data?

Comment (tarikeshaq) [reply Storage/SQL/SQLiteHistory.swift:173]: The plan is to keep the data for long so we can monitor the sizes of databases and compare this implementation and the application-services one. But we will be getting rid of this method when we swap to using application services history

That said, also with the application services history we will want to keep track of the number of visits, but that will probably happen in Rust as opposed to here in the app

**Review by nbhasin2 — APPROVED** [review]

File: Storage/SQL/SQLiteHistory.swift
Comment (nbhasin2) [reply Storage/SQL/SQLiteHistory.swift:173]: @tarikeshaq good to merge, let me know if you have proper permission

Comment (tarikeshaq) [reply Storage/SQL/SQLiteHistory.swift:173]: @nbhasin2 thanks a ton!!! unfortunely I don't :( Is there a way I can get the permission to land a PR once it's approved?

Comment (nbhasin2) [reply Storage/SQL/SQLiteHistory.swift:173]: Merged!!

**Pull request merged**
